Noncanonical mappend

These warnings should be fixed at some point:
```
Building library for indexed-traversable-0.1.2..
...
[2 of 5] Compiling WithIndex        ( src/WithIndex.hs, dist/build/WithIndex.o, dist/build/WithIndex.dyn_o )

src/WithIndex.hs:673:3: warning: [-Wnoncanonical-monoid-instances]
    Noncanonical ‘mappend’ definition detected
    in the instance declaration for ‘Monoid (Traversed a f)’.
    ‘mappend’ will eventually be removed in favour of ‘(<>)’
    Either remove definition for ‘mappend’ (recommended) or define as ‘mappend = (<>)’
    See also: https://gitlab.haskell.org/ghc/ghc/-/wikis/proposal/semigroup-monoid
    |
673 |   Traversed ma `mappend` Traversed mb = Traversed (ma *> mb)
    |   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/WithIndex.hs:694:3: warning: [-Wnoncanonical-monoid-instances]
    Noncanonical ‘mappend’ definition detected
    in the instance declaration for ‘Monoid (Sequenced a m)’.
    ‘mappend’ will eventually be removed in favour of ‘(<>)’
    Either remove definition for ‘mappend’ (recommended) or define as ‘mappend = (<>)’
    See also: https://gitlab.haskell.org/ghc/ghc/-/wikis/proposal/semigroup-monoid
    |
694 |   Sequenced ma `mappend` Sequenced mb = Sequenced (ma >> mb)
    |   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

--- a/indexed-traversable/src/WithIndex.hs
+++ b/indexed-traversable/src/WithIndex.hs
@@ -670,7 +670,7 @@ instance Applicative f => Semigroup (Traversed a f) where
 instance Applicative f => Monoid (Traversed a f) where
   mempty = Traversed (pure (error "Traversed: value used"))
   {-# INLINE mempty #-}
-  Traversed ma `mappend` Traversed mb = Traversed (ma *> mb)
+  mappend = (<>)
   {-# INLINE mappend #-}
 
 ------------------------------------------------------------------------------
@@ -691,7 +691,7 @@ instance Monad m => Semigroup (Sequenced a m) where
 instance Monad m => Monoid (Sequenced a m) where
   mempty = Sequenced (return (error "Sequenced: value used"))
   {-# INLINE mempty #-}
-  Sequenced ma `mappend` Sequenced mb = Sequenced (ma >> mb)
+  mappend = (<>)
   {-# INLINE mappend #-}
 
 ------------------------------------------------------------------------------

